pkg/ipmi/ipmi.go

```diff
@@ -77,6 +82,34 @@ type systemInterfaceAddr struct {
 	lun      byte //nolint:unused
 }
 
+type StandardEvent struct {
+	Timestamp    uint32
+	GenID        uint16
+	EvMRev       uint8
+	SensorType   uint8
+	SensorNum    uint8
+	EventTypeDir uint8
+	EventData    [3]uint8
+}
+
+type OEMTsEvent struct {
+	Timestamp        uint32
+	ManfID           [3]uint8
+	OEMTsDefinedData [6]uint8
+}
+
+type OEMNontsEvent struct {
+	OEMNontsDefinedData [13]uint8
+}
+
+type Event struct {
+	RecordID      uint16
+	RecordType    uint8
+	StandardEvent //3:15
```

Comment:
This is trying to emulate the "c" unions. I wonder if there is a better alternative, maybe having an event interface.

@morganjangwiwynn Do we plan to use the 3 record types from u-root?

Reply:
@pmazzini At least two record types will be used so far, standard and OEM non-timestamped.

Reply:
I implemented an event interface.

Reply:
Sorry for changing my mind. I think the previous implementation was better but with an additional `Marshal` method.

Example:

```
type Event struct {
	RecordID      uint16
	RecordType    uint8
        StandardEvent //3:15
        ...
}

type StandardEvent struct {
        ...
}

func (e *Event) marshall() bytes {
        ...
}

And in that way the IPMI method does not need to know how to convert the event into bytes.
```

Reply:
I implemented the marshal method for converting event to bytes.